drivers/amazonec2/awscredentials.go

```diff
@@ -1,19 +1,79 @@
 package amazonec2
 
-import "github.com/aws/aws-sdk-go/aws/credentials"
+import (
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
 
 type awsCredentials interface {
-	NewStaticCredentials(id, secret, token string) *credentials.Credentials
+	Credentials() *credentials.Credentials
+}
+
+type ProviderFactory interface {
+	NewStaticProvider(id, secret, token string) credentials.Provider
+
+	NewSharedProvider(filename, profile string) credentials.Provider
+}
+
+type defaultAWSCredentials struct {
+	AccessKey        string
+	SecretKey        string
+	SessionToken     string
+	Filename         string
+	Profile          string
+	providerFactory  ProviderFactory
+	fallbackProvider awsCredentials
+}
+
+func NewAWSCredentials(id, secret, token, filename, profile string) *defaultAWSCredentials {
```

Comment:
Easily. I kept them in there to maintain the previous behavior. Since there is no external way for those to be updated, that code can be killed. I'll do that before I resubmit on top of pull #3772.